test/test_evaluation/test_train_evaluator.py

```diff
@@ -1176,6 +1221,50 @@ def test_get_splitter_cv_object(self, te_mock):
         with self.assertRaises(ValueError):
             evaluator.get_splitter(D)
 
+        # GroupKFold, regression with args
+        D.data['Y_train'] = np.array([0.0, 0.1, 0.2, 0.3, 0.4, 0.5])
+        D.info['task'] = REGRESSION
+        evaluator = TrainEvaluator()
+        evaluator.resampling_strategy = GroupKFold
+        evaluator.resampling_strategy_args = {'folds': 2, 'groups': np.array([1, 1, 2, 1, 2, 2])}
+        cv = evaluator.get_splitter(D)
+        self.assertIsInstance(cv, GroupKFold)
+        self.assertEqual(cv.get_n_splits(groups=evaluator.resampling_strategy_args['groups']), 2)
+        next(cv.split(D.data['Y_train'], D.data['Y_train'],
+                      groups=evaluator.resampling_strategy_args['groups']))
+
+        # GroupKFold, regression no args
+        D.data['Y_train'] = np.array([0.0, 0.1, 0.2, 0.3, 0.4, 0.5])
+        D.info['task'] = REGRESSION
+        evaluator = TrainEvaluator()
+        evaluator.resampling_strategy = GroupKFold
+        evaluator.resampling_strategy_args = None
+        with self.assertRaises(ValueError):
+            evaluator.get_splitter(D)
+
+        # GroupKFold, multi-output regression with args
+        D.data['Y_train'] = np.array([[0.0, 0.1], [0.2, 0.3], [0.4, 0.5],
+                                     [1.0, 1.1], [1.2, 1.3], [1.4, 1.5]])
+        D.info['task'] = MULTIOUTPUT_REGRESSION
+        evaluator = TrainEvaluator()
+        evaluator.resampling_strategy = GroupKFold
+        evaluator.resampling_strategy_args = {'folds': 2, 'groups': np.array([1, 1, 2, 1, 2, 2])}
+        cv = evaluator.get_splitter(D)
+        self.assertIsInstance(cv, GroupKFold)
+        self.assertEqual(cv.get_n_splits(groups=evaluator.resampling_strategy_args['groups']), 2)
+        next(cv.split(D.data['Y_train'], D.data['Y_train'],
+                      groups=evaluator.resampling_strategy_args['groups']))
+
+        # GroupKFold, multi-output regression no args
+        D.data['Y_train'] = np.array([[0.0, 0.1], [0.2, 0.3], [0.4, 0.5],
+                                     [1.0, 1.1], [1.2, 1.3], [1.4, 1.5]])
+        D.info['task'] = MULTIOUTPUT_REGRESSION
+        evaluator = TrainEvaluator()
+        evaluator.resampling_strategy = GroupKFold
+        evaluator.resampling_strategy_args = None
+        with self.assertRaises(ValueError):
+            evaluator.get_splitter(D)
```

Comment:
same here.

Reply:
Thanks, I changed detect value errors with regex as the error message provided in the train_evaluator.py. Is it correct to do it like this?

Reply:
That's exactly how it should look (at least in my opinion ;) - I prefer to have it that way so we know exactly what we're testing for)